doc/setting_up_sidewalk_environment/setting_up_sidewalk_product.rst

```diff
@@ -42,27 +46,70 @@ Generating provisioning
 
 #. Run device provisioning scripts:
 
-   .. code-block:: console
+   a. Setup Python virtual environment for the provisioning tools
+
+      .. code-block:: console
+
+         $ python3 -m pip install --user virtualenv
+         $ python3 -m venv sample-app-env
+         $ source sample-app-env/bin/activate
+         (sample-app-env)$ python3 -m pip install --upgrade pip
+         (sample-app-env)$ python3 -m pip install -r requirements.txt
+
+   #. Run device provisioning scripts:
+
+      .. code-block:: console
+   
+         (sample-app-env)$ python3 EdgeDeviceProvisioning/provision_sidewalk_end_device.py
+
+      You should see the following output:
+
+      .. code-block:: console
 
-      $ python3 EdgeDeviceProvisioning/provision_sidewalk_end_device.py
+         INFO:root:Status: 200
+         INFO:root:Saving wireless device to file
+         INFO:root:Generating MFG by calling provision.py
+         INFO:root:  Generating MFG.hex for Nordic
+         INFO:root:Done!
 
-   You should see the following output:
+   #. Exit Python virtual environment
+
+      .. code-block:: console
+
+         (sample-app-env)$ deactivate
+
+#. Flash the :file:`Nordic_MFG.hex` file:
+
+   Your provisioning file is located in `EdgeDeviceProvisioning` folder.
+   Devices are grouped in device profiles folders as shown in structure below:
 
    .. code-block:: console
 
       EdgeDeviceProvisioning \
-      - DeviceProfile_102d750c-e4d0-4e10-8742-ea3698429ca9 \
+      - DeviceProfile_<profile-id> \
          - DeviceProfile.json
-         - WirelessDevice_5153dd3a-c78f-4e9e-9d8c-3d84fabb8911\
+         - WirelessDevice_<device-id>\
              --  Nordic_MFG.bin
              --  Nordic_MFG.hex
              --  WirelessDevice.json
 
-#. Flash the :file:`Nordic_MFG.hex` file:
+   a. Go to device folder
 
-   .. code-block:: console
+      .. code-block:: console
+
+         $ cd EdgeDeviceProvisioning/DeviceProfile_<profile-id>/WirelessDevice_<device-id>
+
+      For example:
+
+      .. code-block:: console
+
+         $ cd EdgeDeviceProvisioning/DeviceProfile_102d750c-e4d0-4e10-8742-ea3698429ca9/WirelessDevice_5153dd3a-c78f-4e9e-9d8c-3d84fabb8911
```

Comment:
I have a proposal to add additional info that the pattern after `WirelessDevice_` subcatalog name is a device ID which can be used with MQTT to send messages to device. It can be helpful for user.

Reply:
Added a reference in testing document. I point to the `WirelessDevice.json` file instead of folder name